File: 프로그래머스/Lv.1/12943. 콜라츠 추측/콜라츠 추측.cs
public class Solution {
    public int solution(int num) {
        int answer = 0;
        int temp = num;
        while(temp != 1)
        {
            if(temp % 2 == 0)
            {
                temp = temp / 2;
            }
            else if(temp % 2 == 1)
            {
                temp = (temp * 3) + 1; 
            }
            answer++;
            
            if(answer > 500)
            {
                answer = -1;
                break;
            }
        }
        if(num == 1)
            answer = 0;
        return answer;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        long longNum = num;
        int count = 0;
        int attemptCount = 500;
        while(count < attemptCount)
        {
            if(longNum == 1)
                break;
            
            if(longNum % 2 == 0)
            {
                longNum /= 2;
            }
            else
            {
                longNum = longNum * 3 + 1;
            }
            
            count++;
        }
        return count >= attemptCount ? -1 : count;
    }
}